addons/resource/models/res_users.py

```diff
@@ -12,3 +12,12 @@ class ResUsers(models.Model):
     resource_calendar_id = fields.Many2one(
         'resource.calendar', 'Default Working Hours',
         related='resource_ids.calendar_id', readonly=False)
+
+    def write(self, vals):
+        rslt = super().write(vals)
+
+        # If the timezone of the admin user gets set on their first login, also update the timezone of the default working calendar
+        if vals.get('tz') and len(self) == 1 and not self.env.user.log_ids and self.env.user == self.env.ref('base.user_admin'):
```

Comment:
```suggestion
        if vals.get('tz') and len(self) == 1 and not self.env.user.login_date and self.env.user == self.env.ref('base.user_admin'):
```

Reply:
I'm not sure, but maybe you should also check that `self == self.env.user` ? :thinking:

Reply:
I don't know if there are situations where it could make a difference, but it seems sensible to add it just in case :+1: